src/util/InteractionController.ts

```diff
@@ -0,0 +1,29 @@
+import { trimTrailingSlashes } from './Util';
+import { v4 as uuid } from 'uuid';
+import { LINK_TYPE_LDP_BC, LINK_TYPE_LDPC } from './LinkTypes';
+
+export class InteractionController {
+  /**
+   * Check whether a new container or a resource should be created based on the given parameters.
+   * @param slug - Incoming slug header.
+   * @param link - Incoming link header.
+   */
+  public isContainer(slug?: string, link?: Set<string>): boolean {
+    if (!slug || !slug.endsWith('/')) {
+      return link !== undefined && (link.has(LINK_TYPE_LDPC) || link.has(LINK_TYPE_LDP_BC));
```

Comment:
Usually, we never compare to `undefined` in JavaScript (we do a `typeof` check). I'd consider `?.has` here, or just `link && (`.

Reply:
In both cases, I get a `TS2322: Type 'boolean | undefined' is not assignable to type 'boolean'.   Type 'undefined' is not assignable to type 'boolean'.`.
I can change it into `return link?.has(LINK_TYPE_LDPC) as boolean || link?.has(LINK_TYPE_LDP_BC) as boolean;`
or `return typeof link !== 'undefined' && (link.has(LINK_TYPE_LDPC) || link.has(LINK_TYPE_LDP_BC));`.
Which one do you prefer, or do you have a nicer alternative?

Reply:
`!!link &&`? 🙂

But then I guess the linter will complain about our casting to boolean. Ugh.

`typeof link !== 'undefined'` if we have to

Reply:
The correct way (well according to the linter that is), is wrapping it in a `Boolean` call. `Boolean(undefined)` is valid.

Reply:
That would be `return Boolean(link?.has(LINK_TYPE_LDPC)) || Boolean(link?.has(LINK_TYPE_LDP_BC));` which is more or less the same as my first option, is that okay?

Reply:
Or can we `Boolean(link?.has(LINK_TYPE_LDPC || link?.has(LINK_TYPE_LDP_BC))`

Reply:
Unfortunately not, then the linter complains: `ESLint: Prefer using nullish coalescing operator (`??`) instead of a logical or (`||`), as it is a safer operator.(@typescript-eslint/prefer-nullish-coalescing)` and with `??` instead of `||` it gives wrong results.

Reply:
JavaScript in 2020, ladies and gents 😅
I'm fine with any solution, let's just not equality check for `undefined` (which is a re-definable variable).